extensions/qute/runtime/src/main/java/io/quarkus/qute/debug/DebugQuteEngineObserver.java

```diff
@@ -0,0 +1,54 @@
+package io.quarkus.qute.debug;
+
+import jakarta.annotation.PreDestroy;
+import jakarta.enterprise.event.Observes;
+import jakarta.inject.Singleton;
+
+import io.quarkus.qute.EngineBuilder;
+import io.quarkus.qute.debug.adapter.RegisterDebugServerAdapter;
+import io.quarkus.qute.runtime.QuteConfig;
+import io.quarkus.runtime.LaunchMode;
+
+/**
+ * Observes the creation of Qute engines and attaches the Qute debugger in development mode.
+ *
+ * <p>
+ * When a new {@link io.quarkus.qute.EngineBuilder} is observed and the application is running in
+ * {@link io.quarkus.runtime.LaunchMode#DEVELOPMENT development mode} with
+ * {@code quarkus.qute.debug.enabled=true}, this observer:
+ * <ul>
+ * <li>Enables template tracing on the engine (required for the debugger).</li>
+ * <li>Registers a {@link RegisterDebugServerAdapter} to allow DAP clients to connect.</li>
+ * </ul>
+ *
+ * <p>
+ * The {@link #cleanup()} method ensures that the debugger is properly reset when the application shuts down.
+ */
+@Singleton
+public class DebugQuteEngineObserver {
```

Comment:
This class should go to the `io.quarkus.qute.runtime.debug` package. Otherwise it could be treated as public API.

I would also annotate the class with `@io.smallrye.common.annotation.Experimental`.

Reply:
Done